rule/unsecure_url_scheme.go

```diff
@@ -0,0 +1,80 @@
+package rule
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strconv"
+	"strings"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// UnsecureURLSchemeRule checks if a file contains string literals with unsecure URL schemes (for example: http://... in place of https://...).
+type UnsecureURLSchemeRule struct{}
+
+// Apply applied the rule to the given file.
+func (*UnsecureURLSchemeRule) Apply(file *lint.File, _ lint.Arguments) []lint.Failure {
+	if file.IsTest() {
+		return nil // skip test files
+	}
+
+	var failures []lint.Failure
+	onFailure := func(failure lint.Failure) {
+		failures = append(failures, failure)
+	}
+
+	w := lintUnsecureURLSchemeRule{
+		onFailure: onFailure,
+	}
+
+	ast.Walk(w, file.AST)
+	return failures
+}
+
+// Name returns the rule name.
+func (*UnsecureURLSchemeRule) Name() string {
+	return "unsecure-url-scheme"
+}
+
+type lintUnsecureURLSchemeRule struct {
+	onFailure func(lint.Failure)
+}
+
+func (w lintUnsecureURLSchemeRule) Visit(node ast.Node) ast.Visitor {
+	n, ok := node.(*ast.BasicLit)
+	if !ok || n.Kind != token.STRING {
+		return w // not a string litereal
+	}
+
+	value, _ := strconv.Unquote(n.Value) // n.Value has one of the following forms: "..." or `...`
+
+	var scheme string
+	var urlPrefixLen int
+	switch {
+	case strings.HasPrefix(value, `http:/`):
+		scheme = "http"
+		urlPrefixLen = 7 // http://
+	case strings.HasPrefix(value, `ws:/`):
```

Comment:
I don't get why you are detecting the http:/ and ws:/ prefixes without a double /

Reply:
I feel like it's my fault

- https://github.com/mgechev/revive/pull/1478#discussion_r2275954555

There was a typo in my comment

I meant http://::/ and http://::1

 Not with a simple slash

🤦‍♂️😒